### PR TITLE
doc: extend conduct document to parallel golang.org/conduct

### PR DESCRIPTION
**Pull request by kortschak**
This adds a complete Code of Conduct document for the Gonum project. The starting point for the document was the Go Community Code of Conduct since this was the basis for the original conduct document we had and by keeping the the Gonum document similar to the Go document simplifies the space.

The differences here mainly reflect that the sponsors will be NumFOCUS rather than Google and that Gonum is likely not yet large enough to need us to have a distinct Project Steward role in addition to the conduct committee (see https://blog.golang.org/conduct-2018 for a discussion of the changes the Go project made from a committee to a stewarding model backed by a project-independent committee).

We can revisit the model as needed. I will send a post to gonum-dev notifying people of this and the background for it so that other interested people can comment. Probably also the Gophers #data-science channel.

Please take a look.

### CODE REVIEW

**Comment by kortschak** [comment]
I have not put the names of people in the committee at this stage as we have not yet decided who they should be. This can either be an additional commit here or another PR.

**Comment by kortschak** [comment]
Waiting on feedback from NumFOCUS and seeing if we can get some additional candidate committee members before I go ahead with adding names and merging.

**Comment by kortschak** [comment]
PTAL

File: CONDUCT.md [hunk 1]
```diff
@@ -1,12 +1,94 @@
...
+
+The first goal of the Code of Conduct is to specify a baseline standard of behavior so that people with different social values and communication styles can talk about Gonum effectively, productively, and respectfully.
+
+The second goal is to provide a mechanism for resolving conflicts in the community when they arise.
+
+The third goal of the Code of Conduct is to make our community welcoming to people from different backgrounds. Diversity is critical to the project; for Gonum to be successful, it needs contributors and users from all backgrounds. (See [Go, Open Source, Community.](https://blog.golang.org/open-source))
+
+We believe that healthy debate and disagreement are essential to a healthy project and community. However, it is never ok to be disrespectful. We value diverse opinions, but we value respectful behavior more.
```
Comment (romanwerpachowski) [inline CONDUCT.md:13]: I wouldn't use "ok" in a formal document. It should be at least spelled "OK", or maybe changed to "acceptable" (though "OK" sounds best to me personally).

File: CONDUCT.md [hunk 2]
```diff
@@ -1,12 +1,94 @@
...
+	- Discussing potentially offensive or sensitive issues; this all too often leads to unnecessary conflict.
+	- Microaggressions: brief and commonplace verbal, behavioral and environmental indignities that communicate hostile, derogatory or negative slights and insults to a person or group. 
+
+People are complicated. You should expect to be misunderstood and to misunderstand others; when this inevitably occurs, resist the urge to be defensive or assign blame. Try not to take offense where no offense was intended. Give people the benefit of the doubt. Even if the intent was to provoke, do not rise to it. It is the responsibility of *all parties* to de-escalate conflict when it arises.
+
+## Code of Conduct
+
+### Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.
```
Comment (romanwerpachowski) [inline CONDUCT.md:43]: Why is this list slightly different than the list at the top of the file? Could we use this one (which is more complete) in both places?

File: CONDUCT.md [hunk 3]
```diff
@@ -1,12 +1,94 @@
...
+
+This Code of Conduct also applies outside the project spaces when the Code of Conduct Committee have a reasonable belief that an individual's behavior may have a negative impact on the project or its community.
+
+### Conflict Resolution
+
+We do not believe that all conflict is bad; healthy debate and disagreement often yield positive results. However, it is never okay to be disrespectful or to engage in behavior that violates the project's code of conduct.
+
+If you see someone violating the code of conduct, you are encouraged to address the behavior directly with those involved. Many issues can be resolved quickly and easily, and this gives people more control over the outcome of their dispute. If you are unable to resolve the matter for any reason, or if the behavior is threatening or harassing, report it. We are dedicated to providing an environment where participants feel welcome and safe.
```
Comment (romanwerpachowski) [inline CONDUCT.md:79]: "this Code of Conduct" IMHO

**Comment by kortschak** [comment]
@romanwerpachowski Sorry, I should have made this clear. The request for approval was for adding your name. The text is essentially the same as the Go Code of Conduct and has already been accepted.

**Comment by romanwerpachowski** [comment]
I see :) Never mind then, it was just a few nitpicks.

**Comment by kortschak** [comment]
@nicole-numfocus This has been merged and we have also added a more [public-facing copy](https://www.gonum.org/conduct/) at the website, discoverable via the [Help menu/page](https://www.gonum.org/help/).

**Comment by nicole-numfocus** [comment]
Thanks for the heads up @kortschak. I'll let the appropriate parties know we are ready to move forward with onboarding Gonum. 🙂

**Pull request merged**